### DISCUSSION TITLE
When using GoogleProvider invalid_client (The OAuth client was not found.) - issue with openid-client v 5.6.3

### DISCUSSION THREAD

**Discussion by bra1nDump** [nextauthjs/next-auth#9585] [Help]
It seems like I triggered an issue in a dependency of `next-auth` - `openid-client`
Broken version `openid-client` for GoogleProvider: `5.6.3`
Upgraded to version: `5.6.4` and the problem was resolved.

The issue suggests more providers might be broken that rely on openid client. Not sure if GithubProvider does, but it was working fine.

Version `5.6.3` introduced a bug with a lot of providers as per `5.6.4` release. https://github.com/panva/node-openid-client/releases/tag/v5.6.4

I was using stock GoogleProvider and it was giving me this error
```log
[next-auth][error][OAUTH_CALLBACK_ERROR] 
https://next-auth.js.org/errors#oauth_callback_error invalid_client (The OAuth client was not found.) {
  error: OPError: invalid_client (The OAuth client was not found.)
      at processResponse (webpack-internal:///(rsc)/./node_modules/openid-client/lib/helpers/process_response.js:34:19)
      at Client.grant (webpack-internal:///(rsc)/./node_modules/openid-client/lib/client.js:1206:28)
      at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
      at async Client.callback (webpack-internal:///(rsc)/./node_modules/openid-client/lib/client.js:418:30)
      at async oAuthCallback (webpack-internal:///(rsc)/./node_modules/next-auth/core/lib/oauth/callback.js:120:22)
      at async Object.callback (webpack-internal:///(rsc)/./node_modules/next-auth/core/routes/callback.js:18:79)
      at async AuthHandler (webpack-internal:///(rsc)/./node_modules/next-auth/core/index.js:202:38)
      at async NextAuthRouteHandler (webpack-internal:///(rsc)/./node_modules/next-auth/next/index.js:50:30)
      at async NextAuth._args$ (webpack-internal:///(rsc)/./node_modules/next-auth/next/index.js:85:24)
      at async <redacted> /node_modules/next/dist/compiled/next-server/app-route.runtime.dev.js:6:63251 {
    name: 'OAuthCallbackError',
    code: undefined
  },
  providerId: 'google',
  message: 'invalid_client (The OAuth client was not found.)'
}
```

**Comment by TahreemZafar** [comment]
Hi! I'm getting the same error as above in my terminal's console. I have tried to change the `openid-client` dependency's version in the **node modules** of my project from 5.6.3 to 5.6.4 , but I'm still getting the same error. I had also tried many other things like rechecking the `clientId` and `clientSecret` , removing string quotes and spaces from id and secret in the env file and as well as deleting the old credentials and creating new ones again and again but still its not working.
 I am quite a beginner in all those error handling stuff. So it would be of great help if someone can simplify the solution.
     Thanks...


In the browser this is what I'm getting...

![1](https://github.com/nextauthjs/next-auth/assets/105108420/ba85c281-cfc1-4c54-a8d2-f4c20bfa9deb)

**Comment by TahreemZafar** [comment]
**Back Again**

I have found the solution. If you're still getting the error after upgrading the version than try this.

Go and save your current project somewhere and create a new project folder. And generate a new `create-next-app` and install the latest `next-auth` dependency. Than copy all of your code into the new project except the `node modules`.

The error above is only occurring in the old `create-next-app`'s dependencies. You won't have this error if you are creating the latest ones.

**Reply by kurays015** [reply]
![image](https://github.com/nextauthjs/next-auth/assets/126612073/4e26c9ea-5bfc-4fc5-8fda-7ab227c57bdf)
I encountered the error with this version.

**Reply by MingyuJeon** [reply]
@kurays015 Could you resolve this issue?
I'm stuck on this issue as well.

**Reply by kurays015** [reply]
@MingyuJeon  I didn't actually solve it. I just changed my next-auth version to v5 beta and it's working again.

**Reply by MingyuJeon** [reply]
@kurays015 Thank you so much for letting me know.
But your next version is 14.0.4 still?
I'm using 14.1.4 and ^5.0.0-beta.16 but it's not working :/

**Reply by kurays015** [reply]
@MingyuJeon yes 14.1.4, the set up in v5 is different from v4 
https://authjs.dev/guides/upgrade-to-v5

**Reply by zNetinho** [reply]
Good morning, the keys in your .env.local file have the prefix "AUTH_...'?
![image](https://github.com/nextauthjs/next-auth/assets/95658733/badce435-68f3-43cf-bf9e-193949747108)

**Reply by 2-fly-4-ai** [reply]
> Good morning, the keys in your .env.local file have the prefix "AUTH_...'? ![image](https://private-user-images.githubusercontent.com/95658733/346911660-badce435-68f3-43cf-bf9e-193949747108.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE3OTE3OTcsIm5iZiI6MTcyMTc5MTQ5NywicGF0aCI6Ii85NTY1ODczMy8zNDY5MTE2NjAtYmFkY2U0MzUtNjhmMy00M2NmLWJmOWUtMTkzOTQ5NzQ3MTA4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzI0VDAzMjQ1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTYzM2M5MTgyNjk2M2JhNDE2ZDI5NTFjNTYyMmRkNGVmNGRjMmU4OGJhZGNiNWZiYzg2MzMyNGE5NWZjZTJmZWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.3oEeD_Xz6_K1n3CzNeH90a6HXyXh7UUKI1Hveb9G08Q)

Since when?
